Lib/test/test_types.py

```diff
@@ -768,6 +769,13 @@ def test_iterators(self):
         self.assertEqual(set(view.values()), set(values))
         self.assertEqual(set(view.items()), set(items))
 
+    def test_reversed(self):
+        d = {'a': 1, 'b': 2, 'foo': 0, 'c': 3, 'd': 4}
+        del d['foo']
+        r = reversed(self.mappingproxy(d))
```

Comment:
I think creating the proxy *before* deleting the element makes sure that this is indeed an updating view:

```suggestion
        mp = self.mappingproxy(d)
        del d['foo']
        r = reversed(mp)
```